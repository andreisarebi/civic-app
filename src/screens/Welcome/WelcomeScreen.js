import React from 'react';
import PropTypes from 'prop-types';
import { Image, StyleSheet, View } from 'react-native';
import colors from '../../styles/colors';
import WelcomeCarousel from './WelcomeCarousel';
import WelcomePanel from './WelcomePanel';

const civicLogoIcon = require('../../assets/images/civic-logo-white.png');

const WelcomeScreen = props => {
  return (
    <View style={styles.container}>
      <View></View>
      <View style={styles.mastheadContainer}>
        <Image style={styles.civicLogo} source={civicLogoIcon} />
        <WelcomeCarousel isSmallScreen={props.isSmallScreen} />
      </View>
      <View></View>
      <WelcomePanel
        formType={props.formType}
        switchFormType={props.changeFormType}
        navigate={props.navigation.navigate}
      />
    </View>
  );
};

WelcomeScreen.propTypes = {
  formType: PropTypes.string.isRequired,
  isSmallScreen: PropTypes.bool.isRequired,
  changeFormType: PropTypes.func.isRequired,
  navigation: PropTypes.shape({ navigate: PropTypes.func }).isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: colors.lightBlue,
    paddingTop: 10,
    paddingLeft: 10,
    paddingRight: 10,
  },
  mastheadContainer: {
    alignItems: 'center',
  },
  civicLogo: {
    height: 65,
    width: 178,
    marginBottom: 10,
  },
});

export default WelcomeScreen;
