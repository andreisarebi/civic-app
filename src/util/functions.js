import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
