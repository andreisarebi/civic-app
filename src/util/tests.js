import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
