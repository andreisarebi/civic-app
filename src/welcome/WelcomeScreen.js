import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import colors from '../styles/colors';
import WelcomeCarousel from './WelcomeCarousel';
import WelcomePanel from './WelcomePanel';

const WelcomeScreen = () => {
  return (
    <View style={styles.container}>
      <Image style={styles.civicLogo} source={require('../assets/images/civic-logo-white.png')} />
      <WelcomeCarousel />
      <WelcomePanel />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: colors.lightBlue,
    paddingTop: 10,
    paddingLeft: 10,
    paddingRight: 10,
  },
  civicLogo: {
    height: 65,
    width: 178,
    marginTop: 20,
  },
});

export default WelcomeScreen;
