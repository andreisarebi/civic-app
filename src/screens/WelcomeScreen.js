import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import colors from '../styles/colors';
import Welcome from '../welcome/Welcome';

class WelcomeScreen extends React.Component {
  static propTypes = propTypes;

  render() {
    return (
      <View style={styles.container}>
        <Welcome />
      </View>
    );
  }
}

const propTypes = {
  navigation: PropTypes.objectOf({
    navigate: PropTypes.func,
    push: PropTypes.func,
  }),
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: colors.lightBlue,
  },
});

export default WelcomeScreen;
