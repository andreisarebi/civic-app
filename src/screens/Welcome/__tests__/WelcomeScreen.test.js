import { createSnapshot } from '../../../util/tests';
import WelcomeScreen from '../WelcomeScreen';

describe('WelcomeScreen', () => {
  it('renders correctly', () => {
    const props = {
      formType: 'foo',
      changeFormType: jest.fn(),
      navigation: { navigate: jest.fn() },
      isSmallScreen: false,
    };
    const snapshot = createSnapshot(WelcomeScreen.WrappedComponent, props);
    expect(snapshot).toMatchSnapshot();
  });
});
