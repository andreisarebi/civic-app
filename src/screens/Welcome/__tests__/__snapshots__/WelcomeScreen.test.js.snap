// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2d9cdb",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <Component />
  <Component
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 65,
          "marginBottom": 10,
          "width": 178,
        }
      }
    />
    <WelcomeCarousel
      isSmallScreen={false}
    />
  </Component>
  <Component />
  <WelcomePanel
    formType="foo"
    navigate={[MockFunction]}
    switchFormType={[MockFunction]}
  />
</Component>
`;
